.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,14 @@
+name: pre-commit
+
+on: [push, pull_request]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: '7.0.304'
```

Comment:
I think it might be of little significance, but we usually should use LTS version of dotnet.

Reply:
I'll move github action workflow related stuffs to another PR. I'll reflect it if the option is available that time. (By just sightseeing, gh action runner seems already including latest, not lts version of dotnet..)